pkg/frontend/querymiddleware/results_cache.go

```diff
@@ -232,30 +235,30 @@ func isAtModifierCachable(r Request, maxCacheTime int64, logger log.Logger) bool
 	expr = promql.PreprocessExpr(expr, timestamp.Time(r.GetStart()), timestamp.Time(r.GetEnd()))
 
 	end := r.GetEnd()
-	atModCachable := true
+	cachable := true
+	check := func(ts *int64, offset time.Duration) error {
+		if offset < 0 {
+			cachable = false
```

Comment:
Is this too strict? If we're querying time range `[-36h, -24h]` and negative offset is only `-1h`, can it still be cached?

Reply:
This is too strict, but we agreed offline that we'll allow the negative offsets first, to comply with PromQL, and then will tweak our cache on that (which isn't a big deal since there are very few of those, and we don't support them at all now).

Reply:
Makes sense.